MainWindow.cpp

```diff
@@ -136,10 +138,10 @@ MainWindow::MessageReceived(BMessage* message)
 			BString result;
 			while (message->FindRef("refs", index++, &ref) == B_OK) {
 				Station* station = Station::Load(ref.name, new BEntry(&ref));
-				if ((station = Station::Load(ref.name, new BEntry(&ref)))) {
+				if (station != NULL) {
 					Station* existingStation
 						= fSettings->Stations->FindItem(station->Name());
-					if (existingStation) {
+					if (existingStation != NULL) {
 						delete station;
```

Comment:
I must admit, I need to work on separating my style changes. :)

And hey, no worries! I'm happy to explain:

While it may look like the `station` variable isn't allocated with "new", you'll want to look above at this line:
`Station* station = Station::Load(ref.name, new BEntry(&ref));`

We'll want to look in the Station.cpp file and find the function that's being called above: https://github.com/HaikuArchives/StreamRadio/blob/c654e0c01664830f93567f1cc33e0720f82116a6/Station.cpp#L438

We can see that in this function, that here, a Station object is allocated with "new" here:
https://github.com/HaikuArchives/StreamRadio/blob/c654e0c01664830f93567f1cc33e0720f82116a6/Station.cpp#L443

Eventually, if all goes well in the `Load()` function, we are returned the fresh "Station" object that was created by "new":
https://github.com/HaikuArchives/StreamRadio/blob/c654e0c01664830f93567f1cc33e0720f82116a6/Station.cpp#L551

As a result, the `station` variable is assigned to the Station object that was created with "new".

Reply:
As for the immediate reassignment, you'll want to check out this video about pointers:
https://youtu.be/DTxHyVn0ODg

He also makes a bunch of other great videos on C++ too:
https://www.youtube.com/playlist?list=PLlrATfBNZ98dudnM48yfGUldqGD0S4FFb

Reply:
Thanks for the explanations!
I didn't find anything on "immediate reassignment" in that video. I sat through the 15 minutes of it though and would advise the guy to keep away from the coffee while recording. :)

Reply:
No problem! On the other hand, you get a lot of info and in a short amount of time! ;)